scripts/test/DirectPropertyManagerTest.py

```diff
@@ -1228,8 +1228,56 @@ def test_average_accuracy(self):
         for valExp,valReal in zip(exp_rez,rez):
             self.assertAlmostEqual(valExp,valReal)
 
+    def test_fix_file_extension(self):
+        propman = self.prop_man
+        # Request  files, one really exists, another one does not
+        propman.sample_run = [11001,11111]
+        # put wrong extension (only file with extension .raw exist)
+        propman.data_file_ext = '.nxs'
+        propman.fix_file_extension = False
+        propman.sum_runs = True
+
+        # find file separately. Rinds raw file instead of nxs
+        srd = PropertyManager.sample_run
+        #
+        [ok,file_foud] = srd.find_file(propman,'MAR',11001)
+        self.assertTrue(ok)        
+        self.assertTrue(os.path.isfile(file_foud))
+
+        ok,not_found,found = propman.find_files_to_sum()
+        # verify that if fix_file_extension is False, the system finds
+        # file with extension raw to process further.
+        self.assertFalse(ok)
+        self.assertEqual(len(not_found),1)
+        self.assertEqual(len(found),1)
+        self.assertEqual(not_found[0],11111)
+        self.assertEqual(found[0],11001)
+
+        # clear previously found file extensions, stored in the run list
+        propman.sample_run = []
+        # Request  files, one really exists, another one does not        
+        propman.sample_run = [11001,11111]
+        propman.fix_file_extension = True        
+
+        # find file separately -- impossible due to wrong file extension
+        srd = PropertyManager.sample_run
+        #
+        [ok,message] = srd.find_file(propman,'MAR',11001)
+        self.assertFalse(ok)
+        self.assertEqual(message,'*** Cannot find file named: MAR11001.nxs on Mantid search paths')
+
+
+        ok,not_found,found = propman.find_files_to_sum()
+        # verify that if fix_file_extension is True, the system does not
+        # pick up the file with extension raw.
+        self.assertFalse(ok)
+        self.assertEqual(len(not_found),2)
+        self.assertEqual(len(found),0)
+        self.assertEqual(not_found[0],11001)        
+        self.assertEqual(not_found[1],11111)
+
 
 if __name__ == "__main__":
-    #tester = DirectPropertyManagerTest('test_average_accuracy')
+    #tester = DirectPropertyManagerTest('test_fix_file_extension')
```

Comment:
If this commented out code is not needed then remove it.

Reply:
This one reminds one (me) how to enable single test when debugging Its very useful reminder as in 3 years time, if I return to this piece of code, I will forgot it.

Reply:
This is possible from the command line so this code is not needed, I'll update the developer documentation to reflect this.

In this case it would be `./bin/mantidpython /mantid_src/scripts/test/DirectPropertyManagerTest.py DirectPropertyManagerTest.test_fix_file_extension`

Reply:
I am not running the code from command line during the debug phase. I can probably modify input parameters of the Visual Studio Code to run this particular example but I do not know exactly how  to do it at the moment and not finding these modifications convenient. (Again, may be due to my beginner level with the enviroment) 

An additional way of running things would do no harm